File: src/main/java/it/unibo/crabinv/model/core/GameSnapshot.java
package it.unibo.crabinv.model.core;

import it.unibo.crabinv.model.save.GameSession;

import java.util.List;

/**
 * Contains the data produced by the {@link GameEngine} that needs to be exposed
 * @param renderObjects the list of all game objects to be renderer
 * @param session the {@link GameSession} of the snapshot
 */
public record GameSnapshot (
        List<RenderObjectSnapshot> renderObjects,
        GameSession session
){}

File: src/main/java/it/unibo/crabinv/model/core/snapshot/GameSnapshot.kt
package it.unibo.crabinv.model.core.snapshot

import it.unibo.crabinv.model.core.save.GameSession

/**
 * Contains the data produced by the [GameEngine] that needs to be exposed.
 * @param renderObjects the list of all game objects to be renderer
 * @param session the [GameSession] of the snapshot
 */
@JvmRecord
data class GameSnapshot(
    val renderObjects: MutableList<RenderObjectSnapshot>,
    val session: GameSession?,
)
